src/ompl/geometric/planners/bitstar/BITstar.h

```diff
@@ -0,0 +1,614 @@
+            unsigned int                                             numVerticesDisconnected_;
+
+            /** \brief The number of times a state in the graph was rewired. Accessible via rewiringProgressProperty */
+            unsigned int                                             numRewirings_;
+
+            /** \brief The number of state collision checks. Accessible via stateCollisionCheckProgressProperty */
+            unsigned int                                             numStateCollisionChecks_;
+
+            /** \brief The number of edge collision checks. Accessible via edgeCollisionCheckProgressProperty */
+            unsigned int                                             numEdgeCollisionChecks_;
+
+            /** \brief The number of nearest neighbour calls. Accessible via nearestNeighbourProgressProperty */
+            unsigned int                                             numNearestNeighbours_;
+            ///////////////////////////////////////
+
+            ///////////////////////////////////////
+            //Parameters - Set defaults in construction/setup and DO NOT reset in clear.
+            /** \brief Whether to use a strict-queue ordering (param) */
+            bool                                                     useStrictQueueOrdering_;
+
+            /** \brief The rewiring factor, s, so that r_rrg = s \times r_rrg* > r_rrg* (param) */
+            double                                                   rewireFactor_;
+
+            /** \brief The number of samples per batch (param) */
+            unsigned int                                             samplesPerBatch_;
+
+            /** \brief Track edges that have been checked and failed so they never reenter the queue. (param) */
+            bool                                                     useFailureTracking_;
+
+            /** \brief Option to use k-nearest search for rewiring (param) */
+            bool                                                     useKNearest_;
+
+            /** \brief Whether to use graph pruning (param) */
+            bool                                                     usePruning_;
+
+            /** \brief The fractional decrease in solution cost required to trigger pruning (param) */
+            double                                                   pruneFraction_;
+
+            /** \brief Whether to delay rewiring until a solution is found (param) */
+            bool                                                     delayRewiring_;
+
+            /** \brief Whether to stop the planner as soon as the path changes (param) */
+            bool                                                     stopOnSolnChange_;
+            ///////////////////////////////////////
+        }; //class: BITstar
+    } //geometric
+} //ompl
+
+
+//BIT* Includes:
+//The Vertex ID generator class
+#include "ompl/geometric/planners/bitstar/datastructures/IdGenerator.h"
```

Comment:
I'm not sure if that would work? I had a hard time defining member classes in separate files.

I don't remember all the details, but I know the only way I found to do it was forward decalre the classes in `BITstar` before including the class definitions; otherwise, I would have had these at the top `BITstar.h`...

I guess they'd work in the .cpp, but then when you include BITstar.h somewhere else, you'd be left with incomplete types for types that `BITstar` returns via public functions...